lib/inspector/inspector.dart

```diff
@@ -0,0 +1,181 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library inspector;
+
+import 'dart:async';
+
+import 'package:vm_service_lib/vm_service_lib.dart';
+
+import '../framework/framework.dart';
+import '../globals.dart';
+import '../service_extensions.dart' as extensions;
+import '../ui/custom.dart';
+import '../ui/elements.dart';
+import '../ui/icons.dart';
+import '../ui/primer.dart';
+import '../ui/split.dart';
+import '../ui/ui_utils.dart';
+
+import 'inspector_controller.dart';
+import 'inspector_service.dart';
+import 'inspector_tree_canvas.dart';
+import 'inspector_tree_html.dart';
+import 'inspector_tree_web.dart';
+
+// Generally the canvas tree renderer  is a better fit for the inspector.
+// The html renderer is more appropriate for small static trees such as those
+// generated in the logging view.
+bool _useHtmlInspectorTreeRenderer = false;
+
+// TODO(jacobr): add UI to view and configure the pub root directory.
+
+class InspectorScreen extends Screen {
+  InspectorScreen()
+      : super(
+          name: 'Inspector',
+          id: 'inspector',
+          iconClass: 'octicon-device-mobile',
+        );
+
+  PButton refreshTreeButton;
+
+  SetStateMixin inspectorStateMixin = SetStateMixin();
+  InspectorService inspectorService;
+  InspectorController inspectorController;
+  ProgressElement progressElement;
+  CoreElement inspectorContainer;
+  StreamSubscription<Object> splitterSubscription;
+
+  @override
+  void createContent(Framework framework, CoreElement mainDiv) {
+    mainDiv.add(<CoreElement>[
+      div(c: 'section')
+        ..layoutHorizontal()
+        ..add(<CoreElement>[
+          div(c: 'btn-group')
+            ..add([
+              ServiceExtensionButton(
+                extensions.toggleSelectWidgetMode,
+              ).button,
+              refreshTreeButton =
+                  PButton.icon('Refresh Tree', FlutterIcons.forceRefresh)
+                    ..small()
+                    ..disabled = true
+                    ..click(_refreshInspector),
+            ]),
+          progressElement = ProgressElement()
+            ..clazz('margin-left')
+            ..display = 'none',
+          div()..flex(),
+          div(c: 'btn-group collapsible')
+            ..add(<CoreElement>[
+              ServiceExtensionButton(extensions.performanceOverlay).button,
```

Comment:
possibly put these into ~2 separate button groups, instead of having 7 buttons in a row?

Reply:
done. Split into 3 groups and got a little clever about having the overflow buttons shrink down to just icons before the more important icons
![double-collapse](https://user-images.githubusercontent.com/1226812/51208049-655f2600-18c1-11e9-974d-6a70a95c6d70.gif)